packages/core-data/src/entities.ts

```diff
@@ -216,144 +216,194 @@ export const defaultEntities = [
+	  }
+	| {
+			kind: 'root';
+			name: 'sidebar';
+			recordType: Sidebar< C >;
+			keyType: Sidebar< C >[ EntityKeyName< 'root', 'sidebar' > ];
+			defaultContext: DefaultEntityContext< 'root', 'sidebar' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'widget';
+			recordType: Widget< C >;
+			keyType: Widget< C >[ EntityKeyName< 'root', 'widget' > ];
+			defaultContext: DefaultEntityContext< 'root', 'widget' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'widgetType';
+			recordType: WidgetType< C >;
+			keyType: WidgetType< C >[ EntityKeyName< 'root', 'widgetType' > ];
+			defaultContext: DefaultEntityContext< 'root', 'widgetType' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'user';
+			recordType: User< C >;
+			keyType: User< C >[ EntityKeyName< 'root', 'user' > ];
+			defaultContext: DefaultEntityContext< 'root', 'user' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'comment';
+			recordType: Comment< C >;
+			keyType: Attachment< C >[ EntityKeyName< 'root', 'comment' > ];
+			defaultContext: DefaultEntityContext< 'root', 'comment' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'menu';
+			recordType: NavMenu< C >;
+			keyType: NavMenu< C >[ EntityKeyName< 'root', 'menu' > ];
+			defaultContext: DefaultEntityContext< 'root', 'menu' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'menuItem';
+			recordType: NavMenuItem< C >;
+			keyType: NavMenuItem< C >[ EntityKeyName< 'root', 'menuItem' > ];
+			defaultContext: DefaultEntityContext< 'root', 'menuItem' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'menuLocation';
+			recordType: MenuLocation< C >;
+			keyType: MenuLocation< C >[ EntityKeyName<
+				'root',
+				'menuLocation'
+			> ];
+			defaultContext: DefaultEntityContext< 'root', 'menuLocation' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'navigationArea';
+			recordType: NavigationArea< C >;
+			keyType: NavigationArea< C >[ EntityKeyName<
+				'root',
+				'navigationArea'
+			> ];
+			defaultContext: DefaultEntityContext< 'root', 'navigationArea' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'theme';
+			recordType: Theme< C >;
+			keyType: Theme< C >[ EntityKeyName< 'root', 'theme' > ];
+			defaultContext: DefaultEntityContext< 'root', 'theme' >;
+	  }
+	| {
+			kind: 'root';
+			name: 'plugin';
+			recordType: Plugin< C >;
+			keyType: Plugin< C >[ EntityKeyName< 'root', 'plugin' > ];
+			defaultContext: DefaultEntityContext< 'root', 'plugin' >;
+	  }
+	| {
+			kind: 'postType';
+			name: 'post';
+			recordType: Post< C >;
+			keyType: Post< C >[ 'id' ];
+			defaultContext: 'edit';
+	  }
+	| {
+			kind: 'postType';
+			name: 'page';
+			recordType: Page< C >;
+			keyType: Page< C >[ 'id' ];
+			defaultContext: 'edit';
+	  }
+	| {
+			kind: 'postType';
+			name: 'wp_template';
+			recordType: WpTemplate< C >;
+			keyType: WpTemplate< C >[ 'id' ];
+			defaultContext: 'edit';
+	  }
+	| {
+			kind: 'postType';
+			name: 'wp_template_part';
+			recordType: WpTemplatePart< C >;
+			keyType: WpTemplatePart< C >[ 'id' ];
+			defaultContext: 'edit';
+	  };
```

Comment:
quick note: it's generally convenient to define these separately and then join them together by name. same effect on the types but the names are handy as separate things.

```ts
// entity-type-definitions.ts
export interface Settings { … }

…

export EntityDefinition =
	| Settings
	| Type
	| Attachment
	| …
```

I know it's early and you don't want to focus on the specifics of the type, just wanted to point this out